action/protocol/staking/receipt_log.go

```diff
@@ -55,7 +55,10 @@ func (r *receiptLog) SetData(data []byte) {
 	r.data = data
 }
 
-func (r *receiptLog) Build(ctx context.Context, err error) *action.Log {
+func (r *receiptLog) Build(ctx context.Context, err error, skipLogOnErr bool) *action.Log {
+	if err != nil && skipLogOnErr {
```

Comment:
why add this? don't think it's correct?

for instance, it returns ErrCandidateNotExist (not critical), in this case we should return a receipt with corresponding status

Reply:
BTW, this (skip log on err) is the behavior prior to FbkMigration height

Reply:
fixing the inconsistency of number of topics. skip only after greenland which is larger than fairbankmigrationheight